crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -1274,6 +1274,73 @@ impl<
         self.storage_manager.list_len(&mut self.buf, dst, list);
     }
 
+    fn build_list_with_capacity(
+        &mut self,
+        dst: &Symbol,
+        capacity: Symbol,
+        capacity_layout: InLayout<'a>,
+        ret_layout: &InLayout<'a>,
+    ) {
+        // List alignment argument (u32).
+        let u32_layout = Layout::U32;
+        let list_alignment = self.layout_interner.alignment_bytes(*ret_layout);
+        self.load_literal(
+            &Symbol::DEV_TMP,
+            &u32_layout,
+            &Literal::Int((list_alignment as i128).to_ne_bytes()),
```

Comment:
Should this be to_ne_bytes? I assume it technically should be converted to target architecture bytes?

Reply:
Though maybe I am just mixed up in this api.

Reply:
I followed the structure of `build_list_replace_unsafe`, but I suppose you're right, this would need to follow the target architecture. What would be the proper way we handle this for these methods, is it through `CallConv`?

Reply:
the code is correct, see the implementation of `load_literal`
```rust
    fn load_literal(&mut self, sym: &Symbol, layout: &InLayout<'a>, lit: &Literal<'a>) {
        match (lit, self.layout_interner.get(*layout)) {
            (
                Literal::Int(x),
                Layout::Builtin(Builtin::Int(
                    IntWidth::U8
                    | IntWidth::U16
                    | IntWidth::U32
                    | IntWidth::U64
                    | IntWidth::I8
                    | IntWidth::I16
                    | IntWidth::I32
                    | IntWidth::I64,
                )),
            ) => {
                let reg = self.storage_manager.claim_general_reg(&mut self.buf, sym);
                let val = *x;
                ASM::mov_reg64_imm64(&mut self.buf, reg, i128::from_ne_bytes(val) as i64);
            }
```

so the NE bytes are turned back into an integer here, and then turned into assembly. I suspect this conversion was used to keep the alignment of `Literal` low